src/rebar_config.erl

```diff
@@ -56,35 +56,92 @@ consult_lock_file(File) ->
         [] ->
             [];
         [Locks] when is_list(Locks) -> % beta lock file
-            Locks;
+            read_attrs(beta, Locks, []);
         [{Vsn, Locks}|Attrs] when is_list(Locks) -> % versioned lock file
-            %% Make sure the warning above is to be shown whenever a version
-            %% newer than the current one is being used, as we can't parse
-            %% all the contents of the lock file properly.
-            ?WARN("Rebar3 detected a lock file from a newer version. "
-                  "It will be loaded in compatibility mode, but important "
-                  "information may be missing or lost. It is recommended to "
-                  "upgrade Rebar3.", []),
+            case Vsn of
+                ?CONFIG_VERSION ->
+                    ok;
```

Comment:
There are no earlier lock file versions (only versionless lock files) and there's a match on the format to match only those with a version in there.